diff --git a/js/leaflet.widget.js b/js/leaflet.widget.js
index 0805ce3..ba6f283 100644
--- a/js/leaflet.widget.js
+++ b/js/leaflet.widget.js
@@ -4,6 +4,28 @@
 (function ($, Drupal, drupalSettings) {
   Drupal.behaviors.leaflet_widget = {
     attach: function (context, settings) {
+      $(document).ready(function() {
+        var refresh = function() {
+
+          $.each(settings.leaflet_widget, function (map_id, widgetSettings) {
+            $('#' + map_id, context).each(function () {
+              let map = $(this);
+              let lMap = drupalSettings.leaflet[map_id].lMap;
+
+              // Refreshes map data to load with correct size and bounds
+              lMap.invalidateSize();
+              map.data('leaflet_widget', new Drupal.leaflet_widget(map, lMap, widgetSettings));
+            });
+
+          });              
+        };
+    
+        // Bind refresh function when changing vertical tab.
+        $('.vertical-tabs-list').find('.vertical-tab-button').each(function(key, tab) {
+          $(tab).find('a').bind('click', refresh);
+        });
+      });
+      
       $.each(settings.leaflet_widget, function (map_id, widgetSettings) {
         $('#' + map_id, context).each(function () {
           let map = $(this);
